api/v1alpha3/docluster_types.go

```diff
@@ -29,27 +29,46 @@ const (
 
 // DOClusterSpec defines the desired state of DOCluster.
 type DOClusterSpec struct {
-	// The DigitalOcean Region the cluster lives in.
-	// It must be one of available region on DigitalOcean. See https://developers.digitalocean.com/documentation/v2/#list-all-regions
+	// The DigitalOcean Region the cluster lives in. It must be one of available
+	// region on DigitalOcean. See
+	// https://developers.digitalocean.com/documentation/v2/#list-all-regions
 	Region string `json:"region"`
 	// Network configurations
 	// +optional
 	Network DONetwork `json:"network,omitempty"`
-	// ControlPlaneEndpoint represents the endpoint used to communicate with the control plane.
+	// ControlPlaneEndpoint represents the endpoint used to communicate with the
+	// control plane. If ControlPlaneDNS is unset, the DO load-balancer IP
+	// of the Kubernetes API Server is used.
 	// +optional
 	ControlPlaneEndpoint clusterv1.APIEndpoint `json:"controlPlaneEndpoint"`
+	// ControlPlaneDNS is a managed DNS name that points to the load-balancer
+	// IP used for the ControlPlaneEndpoint.
+	// +optional
+	ControlPlaneDNS *DOControlPlaneDNS `json:"controlPlaneDNS"`
 }
 
 // DOClusterStatus defines the observed state of DOCluster.
 type DOClusterStatus struct {
 	// Ready denotes that the cluster (infrastructure) is ready.
 	// +optional
 	Ready bool `json:"ready"`
+	// ControlPlaneDNSRecordReady denotes that the DNS record is ready and
+	// propagated to the DO DNS servers.
+	// +optional
+	ControlPlaneDNSRecordReady bool `json:"controlPlaneDNSRecordReady,omitempty"`
```

Comment:
Since using `ControlPlaneDNS` is optional, I don't think we should add this status. I would suggest using annotation eg: `docluster.infrastructure.cluster.x-k8s.io/controlplane-dns-propagated: true` to determine is DNS already propagated or not.

Reply:
Why do you think the optionality disqualifies it from the status section?
The status field is marked optional and is omitted if the user does not use this feature. So there should be no confusion if you look at the status of a cluster that is not using this feature.
IMHO the status is a much better fit than an annotation that is updated by a controller.

Reply:
Make sense